libraries/AP_ICEngine/AP_ICEngine.h

```diff
@@ -122,12 +127,20 @@ class AP_ICEngine {
 
     enum class Options : uint16_t {
         DISABLE_IGNITION_RC_FAILSAFE=(1U<<0),
+        DISABLE_REDLINE_WARNING = (1 << 1),
+        DISABLE_REDLINE_GOVERNOR = (1 << 2),
     };
     AP_Int16 options;
 
     // start_chan debounce
     uint16_t start_chan_last_value = 1500;
     uint32_t start_chan_last_ms;
+
+    // redline rpm
+    AP_Int32 redline_rpm;
+    bool redline_flag;
+    float redline_governor_integrator;
+    float redline_throttle_percentage;
```

Comment:
you could be fancy and put these in a `redline` struct.

Reply:
Learned a bit and stumbled around a bit with the IDE warnings. I think what I have now is simple struct.